multi/stm32l4-multi/tty.c

```diff
@@ -124,9 +127,19 @@ static void tty_irqthread(void *arg)
 			ctx->rxready = 0;
 		}
 
-		if (tty_txready(ctx) && libtty_txready(&ctx->tty_common)) {
-			*(ctx->base + tdr) = libtty_getchar(&ctx->tty_common, NULL);
-			*(ctx->base + cr1) |= (1 << 7);
+		if (libtty_txready(&ctx->tty_common)) {
+			if (tty_txready(ctx)) {
```

Comment:
I noticed, that you have changed the order, is this expected?
I think that inlined `tty_txready(ctx)` should be first, before call to libtty, `libtty_txready(&ctx->tty_common)`, thus:
```suggestion
		if (tty_txready(ctx)) {
			if (libtty_txready(&ctx->tty_common)) {
```
or previous, single `if` line.

Reply:
It's ok as is, we need to go to `else` if there is no data in libtty to disable `keepidle`

Reply:
I see, that's ok.